### ISSUE TITLE
Hermite moment g instability: missing dealiasing in RK2 step and RHS assembly

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#122]
## Summary

The Hermite moment field `g` slowly blows up during long simulations due to aliasing error accumulation. The growth rate is ~doubling every 200 steps, worsening at low collisionality where damping cannot compensate. This blocks the collisionality scan (Study 02).

## Root Cause

Two missing dealiasing operations:

### 1. No post-RK2 dealiasing on g (timestepping.py:479-483)

After the full RK2 step with integrating factor:
```python
g_new = apply_streaming_if(
    fields.g + apply_streaming_if(dt * nl_g_half, g_phase_full),
    g_phase_full,
)
# MISSING: g_new = g_new * dealias_mask[:, :, :, jnp.newaxis]
```

The z+/z- fields get implicit dealiasing through the resistive dissipation step (`exp(-eta*k_perp^2r*dt)`), which strongly damps high-k modes. But for g, the resistive dissipation is the same multiplicative factor — it does not enforce a hard cutoff at 2/3 Nyquist.

### 2. No defensive dealiasing in g RHS assembly (physics.py:940, 1016, 1125)

The z+/z- RHS (`z_plus_rhs`, `z_minus_rhs`) applies `* dealias_mask` after every linear combination of brackets (lines 747, 754):
```python
bracket1 = (bracket1a + bracket1b) * dealias_mask  # Defensive
combined_bracket = (bracket1 - bracket2) * dealias_mask  # Defensive
```

But `g0_rhs`, `g1_rhs`, and `gm_rhs` assemble their RHS without re-dealiasing:
```python
rhs = -bracket_phi_g0 - term2 - term3  # No dealias_mask applied!
```

While each individual bracket is dealiased, the linear combination can reintroduce high-k content through floating-point accumulation.

## Why z+/z- Don't Blow Up

The z+/z- nonlinearity is **self-limiting**: z+ advects itself via z- (mutual feedback). Aliasing error in z+ creates nonlinear back-reaction that self-regulates. For g, the advection is by an **external** field Phi = (z+ + z-)/2 — there is no feedback from g back to Phi, so aliasing error grows monotonically.

## RK2 Stability for Advection

The midpoint RK2 amplification factor for pure advection is:
```
|R(iω)|² = 1 + (ω·dt)⁴/4 > 1  for all ω ≠ 0
```

RK2 is unconditionally unstable for advection. For z+/z- this is masked by the self-limiting nonlinearity. For g advected by Phi, the instability is parametric and unbounded.

### Tier 1: Post-RK2 dealiasing (1 line, should fix ~90% of cases)

After line 483 in `timestepping.py`:
```python
g_new = g_new * dealias_mask[:, :, :, jnp.newaxis]
```

### Tier 2: Defensive dealiasing in g RHS (3 functions)

In `g0_rhs`, `g1_rhs`, `gm_rhs`, add `* dealias_mask` after assembling the RHS:
```python
rhs = (-bracket_phi_g0 - term2 - term3) * dealias_mask
```

### Tier 3: Higher-order time integrator (if Tier 1-2 insufficient)

Upgrade to RK4 which has a stability region extending to |ω·dt| ≈ 2.8 on the imaginary axis, vs 0 for RK2.

## Reproduction

Any simulation with M > 0, Lambda != 1, and sufficient runtime will show g growing exponentially. Observable as `eps_nu` (collisional dissipation rate) doubling every ~200 steps. High-nu runs last longer before blowup because collisional damping partially compensates.

## Impact

Blocks Study 02 (collisionality scan) at nu < 1e-3. The first 5 nu values (1e-1 to 1e-3) produce usable data, but lower nu values blow up before the Hermite cascade reaches steady state.
